env_spec.py

```
@@ -19,138 +28,216 @@
-            f'<select id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}">\n'
-        )
-
-        for line in env_var_type_list:
-            if "=" in env_var_type and value == line:
-                ret_str += f'\t<option value="{line}"selected>{line}</option>\n'
-            else:
-                ret_str += f'\t<option value="{line}">{line}</option>\n'
+            line_dict["comment"] = None
 
-        ret_str += "</select>\n"
+        name_match = re.match(name_regex, line)
 
-    return ret_str
+        if name_match:
+            name = name_match.groups()[0]
+            line = name_match.groups()[1]
 
+            ret = re.match(alphanumeric_that_does_not_start_with_digit, name)
 
-def render_env_spec_to_html(input_str):
-    """
-    Takes the whole string and splits it first by \n and then by :, if possible. Returns html output or "".
-    """
-    html_output = ""
+            if ret is None:
+                try:
+                    raise EnvSpecSyntaxError
+                except EnvSpecSyntaxError:
+                    return []
 
-    lines = input_str.split("\n")
+            line_dict["name"] = name
 
-    for line in lines:
-        try:
-            input_field, input_type = line.split(": ")
+            choices_match = re.match(choices_regex, line)
+            default_value_match = re.match(default_values_regex, line)
 
-            if check_input(input_field) and check_type(input_type):
-                html_output += render_env_var_spec(input_field, input_type)
-            else:
-                return ""
-        except:
-            input_field = line
+            if choices_match:
+                choices_str = choices_match.groups()[0]
 
-            if check_input(input_field):
-                html_output += render_env_var_spec(input_field, "")
+                choices = create_list(choices_str)
+
+                if not choices:
+                    try:
+                        raise EnvSpecSyntaxError
```

Can you please raise this exception along with a useful human friendly error?

```py
raise EnvSpecSyntaxError(‘this went wrong and that...`)
```